daemon/image_tag.go

```diff
@@ -13,12 +13,12 @@ func (daemon *Daemon) TagImage(imageName, repository, tag string) error {
 		return err
 	}
 
-	newTag, err := reference.WithName(repository)
+	newTag, err := reference.ParseNormalizedNamed(repository)
```

Comment:
I feel like this function should take a reference instead of assembling it from a string/tag.

Either way, if we switch from `WithName` to `ParseNormalizedNamed`, we should ensure that this is just a name after parsing. I think having a tag or digest inside `repository` should be an error condition, instead of silently being dropped by `TrimNamed` below.